Makefile

```diff
@@ -39,8 +42,14 @@ $(SI_PROTO).tmp: $(SI_SPEC)
 	(diff $@ $(SI_PROTO) > /dev/null 2>&1 || mv -f $@ $(SI_PROTO)) && \
 		rm -f $@
 
+$(CONSTANTS_GO): $(SI_SPEC)
```

Comment:
In the proto spec build we force the build using a temp file. This guarantees a run of the build and then checks if the md document is still in sync with the generated proto file. This does not happen for the constants. Updates to the md document (i.e. a typo fix) could update and replace the constant.go file.
It also causes issues when I revert a change by checking out an older version of the md document. The build is not correctly replacing the generated `constants,go` file. Please setup an execution structure similar to the spec file with a move of a temp file etc:
```
CONSTANTS_GO := constants.go

$(CONSTANTS_GO).tmp: $(SI_SPEC)
	....
	(diff $@ $(PATH_CONSTANTS_GO) > /dev/null 2>&1 || mv -f $@ $(PATH_CONSTANTS_GO)) && \
		rm -f $@
```